src/scripting/StelMainScriptAPI.cpp

```diff
@@ -342,8 +342,29 @@ QStringList StelMainScriptAPI::getAllTimezoneNames()
 
 void StelMainScriptAPI::screenshot(const QString& prefix, bool invert, const QString& dir, const bool overwrite, const QString &format)
 {
+	const bool allowUnsafe = qApp->property("allow_unsafe").toBool();
+	const bool allowed = allowUnsafe || (
+		// Only allow writing in the default directory:
```

Comment:
I have just added path stripping to the `StelMainView::saveScreenShot()`. Only a "naked" filename remains. 

It should be enough to ensure that prefix is not empty (but how risky is a ".tif" file? Oh well, yes, it's hidden on Linux. Just a nameless but visible TIF on Windows.)  and dir is empty.